src/Polly.RateLimiting/RateLimiterRejectedException.cs

```diff
@@ -21,6 +23,19 @@ public RateLimiterRejectedException()
     {
     }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RateLimiterRejectedException"/> class.
+    /// </summary>
+    /// <param name="telemetrySource">The source pipeline and strategy names.</param>
+    public RateLimiterRejectedException(ResilienceTelemetrySource telemetrySource)
```

Comment:
Code analysers recommend a standard set of constructors for publicly-visible types:

- `.ctor()`
- `.ctor(string)`
- `.ctor(string, Exception)`
- `.ctor(SerializationInfo, StreamingContext)`

Because we can't predict what our users might be doing in private code we can't see, we should add at a minimum what someone would need to also populate any new property themselves if they wanted to.

Reply:
Based on this I would say, it is enough for now to introduce those constructors that we need. Future requests asking for other overloads can be easily added. Can we agree that we extend this exception class only with these two overloads?

Reply:
Only those two means you can't create an instance of the exception with all the possible property values, so we would need what you need to make this feature work _plus_ enough extras to allow for a fully constructed object (which could be just one).